### PR TITLE
fix deprecated gha apis

### LINKED ISSUE #1550: fix GHA warnings from usage of ::set-output
**Describe the bug**

See warnings on this run:  https://github.com/sillsdev/web-languageforge/actions/runs/3299222198

**To Reproduce**

We should see these warnings on any run that utilizes deprecated apis

**Expected behavior**

We don't want any warnings if we can avoid them.

**Screenshots**

N/A

**Desktop (please complete the following information):**

N/A

**Mobile (please complete the following information):**

N/A

**Android App in Google Play**

N/A

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by longrunningprocess**
Fixes #1550 

## Description

GHA recently deprecated some apis and this PR migrates to the newer ones.

### Type of Change

- Bug fix (non-breaking change which fixes an issue)

## Screenshots

N/A

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have reviewed the title/description of this PR which will be used as the squashed PR commit message
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works

## How to test

- Double check workflow runs on a pull request and ensure there are no deprecation warnings.
- Double check workflow runs on a a merge to `develop` and ensure there are no deprecation warnings.

## qa.languageforge.org testing

N/A

### CODE REVIEW

**Comment by megahirt** [comment]
Wow!  27 commits - this must have been fun...

File: .github/workflows/build-and-publish-adhoc.yml [hunk 1]
```diff
@@ -1,40 +0,0 @@
-name: Publish ad-hoc image to Docker Hub
```
Comment (megahirt) [inline .github/workflows/build-and-publish-adhoc.yml:1]: Did you intend to delete this file?  I didn't see any mention of deleting this workflow in the PR - is it still useful?

File: .github/workflows/build-and-publish-next.yml [hunk 2]
```diff
@@ -1,37 +0,0 @@
-name: Publish LFNext images to Docker Hub on-demand
```
Comment (megahirt) [inline .github/workflows/build-and-publish-next.yml:1]: Did you intend to delete this file? I didn't see any mention of deleting this workflow in the PR - is it still useful?

**Review by megahirt — COMMENTED** [review]
This PR does more than the description says, including deleting two workflows and bumping a few other GH actions to v3 (not just checkout).

Other than the question about the deleted workflows, this looks good to me.  I can approve once you clarify.  Thanks!

File: .github/workflows/build-and-publish-adhoc.yml (hunk 1 above)
Comment (longrunningprocess) [reply .github/workflows/build-and-publish-adhoc.yml:1]: Yes, it's not used anymore

File: .github/workflows/build-and-publish-next.yml (hunk 2 above)
Comment (longrunningprocess) [reply .github/workflows/build-and-publish-next.yml:1]: It's not useful anymore

**Review by megahirt — APPROVED** [review]
Thanks for clarifying!  Approved

**Comment by megahirt** [comment]
TESTED

I observe no Github warnings or errors in recent GHA builds/tests

**Pull request merged**